core/src/main/java/org/apache/iceberg/SchemaUpdate.java

```diff
@@ -193,6 +204,11 @@ public UpdateSchema renameColumn(String name, String newName) {
       updates.put(fieldId, Types.NestedField.of(fieldId, field.isOptional(), newName, field.type(), field.doc()));
     }
 
+    if (identifierFieldNames.contains(name)) {
```

Comment:
Should we throw exception if the `newName` has been already existed in `identifierFieldNames` (Maybe actually `schema`) ?

Reply:
I agree with this. I think it would make sense to check whether `newName` is already a field in the schema. But that is catching a problem with a rename earlier and isn't really related to this PR.

Reply:
Yeah, I was expecting the lines above to catch this, but surprisingly it did not do this check.